clientv3/client.go

```diff
@@ -503,3 +503,16 @@ func toErr(ctx context.Context, err error) error {
 	}
 	return err
 }
+
+func canceledByCaller(ctx context.Context, err error) bool {
+	if ctx.Err() == nil {
+		return false
+	}
+	if err == nil {
+		return false
+	}
+	if err != context.Canceled && err != context.DeadlineExceeded {
```

Comment:
ah! sorry, I misunderstood your first comment. Yes, your way is much simpler and easy to understand :)